src/Service/ClientService.php

```diff
@@ -108,33 +130,37 @@ public function getClient($salesChannelId)
     }
 
     /**
-     * Get adyen module's version from composer.json
-     * TODO: switch to a shopware service to retrieve the module version instead of composer
+     * Get adyen module's version from PluginEntity
      *
      * @return string
      */
     public function getModuleVersion()
     {
-        $rootDir = $this->containerParametersService->getApplicationRootDir();
-
-        $composerJson = file_get_contents($rootDir . '/custom/plugins/adyen-shopware6/composer.json');
-
-        if (false === $composerJson) {
-            $this->genericLogger->error('composer.json is not available in the Adyen plugin folder');
-            return "NA";
+        try {
+            $moduleVersionCacheItem = $this->cache->getItem(self::MODULE_VERSION_CACHE_TAG);
+        } catch (\Psr\Cache\InvalidArgumentException $e) {
+            $this->genericLogger->error($e->getMessage());
         }
 
-        $composerJson = json_decode($composerJson, true);
-        if (json_last_error() !== JSON_ERROR_NONE) {
-            $this->genericLogger->error('composer.json is not a valid JSON in the Adyen plugin folder');
-            return "NA";
+        //Prefer the cached module version
+        if (!$moduleVersionCacheItem->isHit() || !$moduleVersionCacheItem->get()) {
+            /** @var PluginEntity $pluginEntity */
+            $pluginEntity = $this->pluginRepository->search(
```

Comment:
this is always the installed version right? Is this something we need to keep an eye during testing when we deploy different versions of the code but we don't install it again?

Reply:
Yes this is always the installed version. I don't think it will bring issues with testing as this is only used for the `merchantApplication` field in API requests